ConsoleApp1/Snake.cs

```diff
@@ -0,0 +1,135 @@
+﻿namespace ConsoleApp1
+{
+    public enum Direction
+    {
+        Up,
+        Down,
+        Right,
+        Left
+    }
+    class Snake
+    {
+        private List<Point> _body;
+        private bool _state;
+        private int _boards;
+        private Direction _direction;
+        public bool IsAlive => _state;
+        public Point Head => _body.Last();
+
+        public Snake(int board)
+        {
+            _state = true;
+            _boards = board;
+            _direction = Direction.Right;
+            _body = new List<Point>(3);
+            _body.Add(new Point(1, 1));
+            _body.Add(new Point(2, 1));
+            _body.Add(new Point(3, 1));
+        }
+
+        public void Kill()
+        {
+            _state = false;
+        }
+
+        public void Move(object? source, EventArgs e)
+        {
+            if (_state)
+            {
+                foreach (var bodyPart in _body)
+                {
+                    Console.SetCursorPosition(bodyPart.X, bodyPart.Y);
+                    Console.Write(" ");
+                }
+                _body.RemoveAt(0);
+                CheckMovingDirection();
+                Render();
+                CheckBounds();
+                CheckBodyCollision();
+                ChangeSnakeDirection();
+            }
+        }
+        public void CheckMovingDirection()
+        {
+            (int x, int y) lastCoor = (_body.Last().X, _body.Last().Y);
+            switch (_direction)
+            {
+                case Direction.Up:
+                    _body.Add(new Point(lastCoor.x, lastCoor.y - 1));
+                    break;
+                case Direction.Down:
+                    _body.Add(new Point(lastCoor.x, lastCoor.y + 1));
+                    break;
+                case Direction.Right:
+                    _body.Add(new Point(lastCoor.x + 1, lastCoor.y));
+                    break;
+                case Direction.Left:
+                    _body.Add(new Point(lastCoor.x - 1, lastCoor.y));
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        public void CheckBounds()
```

Comment:
Think about who is responsible for checking bounds, collision and so on

Reply:
added class CollisionObserver